src/Adapters/BlobAdapterUtility.ts

```diff
@@ -0,0 +1,234 @@
+                [event.properties.countWidgetPropertyType]:
+                    visualStats.widgets.propertyCount,
+                [event.properties.countWidgets]: visualStats.widgets.totalCount,
+                [event.properties.parentSceneHash]: sceneHash
+            },
+            triggerType: TelemetryTrigger.SystemAction
+        })
+    );
+}
+
+/** send the KPI telemetry related to individual Elements */
+function sendElementTelemetry(element: IElement, sceneHash: string) {
+    const type = element.type;
+    let meshCount = 0;
+    let aliasCount = 0;
+    // only capture the count when it's actually an 'element'
+    if (ViewerConfigUtility.isTwinToObjectMappingElement(element)) {
+        meshCount = element.objectIDs?.length;
+        if (element.twinAliases) {
+            aliasCount = Object.keys(element.twinAliases).length;
+        }
+    }
+
+    // capture the Element level metrics
+    const event = TelemetryEvents.Adapter.SceneLoad.SystemAction.ParseElement;
+    TelemetryService.sendEvent(
+        new TelemetryEvent({
+            name: event.eventName,
+            customProperties: {
+                [event.properties.countAliases]: aliasCount,
+                [event.properties.countMeshes]: meshCount,
+                [event.properties.parentSceneHash]: sceneHash,
+                [event.properties.elementType]: type
+            },
+            triggerType: TelemetryTrigger.SystemAction
+        })
+    );
+}
+
+/** send the KPI telemetry related to individual Scenes */
+function sendSceneTelemetry(scene: IScene, sceneHash: string) {
+    const assetCount = scene.assets?.length || 0;
+    const behaviorsCount = scene.behaviorIDs?.length || 0;
+    const elementCount = scene.elements?.length || 0;
+    const hasCoordinates =
+        isDefined(scene.latitude) || isDefined(scene.longitude) || false;
+    const hasDescription = scene.description?.trim()?.length > 0;
+    const scenePollingDelay =
+        scene.pollingConfiguration?.minimumPollingFrequency;
+
+    // capture the Scene level metrics
+    const event = TelemetryEvents.Adapter.SceneLoad.SystemAction.ParseScene;
+    TelemetryService.sendEvent(
+        new TelemetryEvent({
+            name: event.eventName,
+            customProperties: {
+                [event.properties.countAssets]: assetCount,
+                [event.properties.countBehaviors]: behaviorsCount,
+                [event.properties.countElements]: elementCount,
+                [event.properties.hasCoordinates]: hasCoordinates,
+                [event.properties.hasDescription]: hasDescription,
+                [event.properties.pollingDelay]: scenePollingDelay,
+                [event.properties.sceneHash]: sceneHash
+            },
+            triggerType: TelemetryTrigger.SystemAction
+        })
+    );
+}
+
+/** send the KPI telemetry related to the Config file as a whole */
+function sendConfigTelemetry(data: I3DScenesConfig) {
```

Comment:
after saving a scene, ParseConfiguration events are fired multiple times, is there a way to make sure they are called once?
<img width="733" alt="image" src="https://user-images.githubusercontent.com/45217314/188204720-3ade17eb-e086-423d-a521-f493143b1cd8.png">

Reply:
In this case you're seeing an event per Element and per Behavior in that scene. That is actually by design. BUT if what you're saying is you don't want that event do go out for the same behavior on each save, then yes, we could throttle it to only send once a day per environment. There's not really any harm though in firing more except that it has more data we have to crawl through. It should be straightforward to eliminate the duplicates in the reports by doing a top 1 by environmentId.